parcels/collectionaos.py

```diff
@@ -0,0 +1,1130 @@
+                for kwvar in kwargs:
+                    if isinstance(kwvar, Field):
+                        continue
+                    if not hasattr(self._data[i], kwvar):
+                        raise RuntimeError('Particle class does not have Variable %s' % kwvar)
+                    setattr(self._data[i], kwvar, kwargs[kwvar][i])
+                    if kwvar not in initialised:
+                        initialised.add(kwvar)
+
+            initialised |= {'lat', 'lon', 'depth', 'time', 'id'}
+
+            for v in self._ptype.variables:
+                if v.name in initialised:
+                    continue
+
+                if isinstance(v.initial, Field):
+                    init_field = v.initial
+                    if init_field.grid.ti != 0:
+                        init_field.fieldset.computeTimeChunk(time[i], 0)
+                    for i in range(self.ncount):
+                        if (time[i] is None) or (np.isnan(time[i])):
+                            raise RuntimeError('Cannot initialise a Variable with a Field if no time provided (time-type: {} values: {}). Add a "time=" to ParticleSet construction'.format(type(time), time))
+                        setattr(self._data[i], v.name, init_field[time[i], depth[i], lat[i], lon[i]])
+                        logger.warning_once("Particle initialisation from field can be very slow as it is computed in scipy mode.")
+
+                if v not in initialised:
+                    initialised.add(v)
+        else:
+            raise ValueError("Latitude and longitude required for generating ParticleSet")
+        self._iterator = None
+        self._riterator = None
+
+    def __del__(self):
+        """
+        Collection - Destructor
+        """
+        super().__del__()
+
+    def iterator(self):
+        self._iterator = ParticleCollectionIteratorAOS(self)
+        return self._iterator
+
+    def __iter__(self):
+        """Returns an Iterator that allows for forward iteration over the
+        elements in the ParticleCollection (e.g. `for p in pset:`).
+        """
+        return self.iterator()
+
+    def reverse_iterator(self):
+        self._riterator = ParticleCollectionIteratorAOS(self, True)
+        return self._riterator
+
+    def __reversed__(self):
+        """Returns an Iterator that allows for backwards iteration over
+        the elements in the ParticleCollection (e.g.
+        `for p in reversed(pset):`).
+        """
+        return self.reverse_iterator()
+
+    def __getitem__(self, index):
+        """
+        Access a particle in this collection using the fastest access
+        method for this collection - by its index.
+
+        :param index: int or np.int32 index of a particle in this collection
+        """
+        return self.get_single_by_index(index)
+
+    def __getattr__(self, name):
+        """
+        Access a single property of all particles.
+        CAUTION: this function is not(!) in-place and is quite slow
```

Comment:
ehm, well, that would be very obstrusive. @RoelBrouwer and me had a long discussion about performance-related hints from those functions, and we decided for the docstring. Otherwise people write their first Jupyter script and the only things he/she sees is a neverending list of warnings. From a user-experience perspective, this way is the nicer one.

Also: precedence. I included some warning in the Dask procedure last year, that indicates that new grids are created because the chunksizes don't match - a performance-related warning. In the end, most people in the group asked (a) what this 'error' means, (b) thought that that error had something to do with crashing code, and (c) rather asked to remove the warning if there is no error going on. Same situation here, but on a larger scale.